### PR TITLE
 Python: generate type stubs for all user functions 

### PR DESCRIPTION
**Pull request by badboy**
After getting asked about it the other day on reddit I figured I give it a go.
Turns out generating the right types in Python is possible and pleases mypy when used in Glean as well (for the most part).

There's some not-yet fixed cases, I want to add some testing and ensure this reliably works, so marking this as WIP for the moment.

### CODE REVIEW

**Review by mhammond — APPROVED** [review]
LGTM but it might make sense to update that comment?

File: uniffi_bindgen/src/interface/types/mod.rs [hunk 1]
```diff
@@ -75,6 +73,10 @@ pub enum Type {
         name: String,
         builtin: Box<Type>,
     },
+    // Types defined in the component API, each of which has a string name.
+    // Defined last, as its functions might reference any other type from above,
+    // so should be defined last.
```
[the remark this answers is not in the document]
Comment (badboy) [reply uniffi_bindgen/src/interface/types/mod.rs:78]: Oh, I think this is not even needed anymore. Let me fix that or update the comment to make it clear.

**Comment by heinrich5991** [comment]
What's the status of this? The changes seem approved. Does it only need a rebase? (Is help with the rebase wanted?)

**Comment by badboy** [comment]
> What's the status of this? The changes seem approved. Does it only need a rebase? (Is help with the rebase wanted?)

Ah yes, this once again needs a rebase. It fell through a bit due to other work, but I can take this up today.

**Pull request merged**